include/flatbuffers/flatbuffers.h

```diff
@@ -697,6 +698,17 @@ class vector_downward {
     }
   }
 
+  void reset() {
+    assert(allocator_ != nullptr);
+    if (buf_ != nullptr) {
+      allocator_->deallocate(buf_, reserved_);
+    }
+    reserved_ = (initial_size_ + sizeof(largest_scalar_t) - 1) &
+                ~(sizeof(largest_scalar_t) - 1);
+    buf_ = allocator_->allocate(reserved_);
```

Comment:
While I'm looking at this, is there a reason for `clear()` or the ctor to allocate? Could we also make that lazy and rely on `make_space(...)` for all allocations? That way an unused or cleared fbb doesnt cost any heap allocations until it's used.